kyuubi-common/src/main/scala/org/apache/kyuubi/config/KyuubiConf.scala

```diff
@@ -2077,6 +2077,24 @@ object KyuubiConf {
       .timeConf
       .createWithDefault(Duration.ofMinutes(30).toMillis)
 
+  val METADATA_CLEANER_BATCH_SIZE: ConfigEntry[Int] =
+    buildConf("kyuubi.metadata.cleaner.batch.size")
+      .serverOnly
+      .doc("The batch size for cleaning expired metadata. " +
+        "This is used to avoid holding the delete lock for a long time " +
+        "when there are too many expired metadata to be cleaned.")
+      .version("1.11.0")
+      .intConf
+      .createWithDefault(Int.MaxValue)
```

Comment:
I expect this is a general perf optimization, should we give a more reasonable default value? e.g. 1000 or 3000